src/EditorFeatures/Core/GoToDefinition/AbstractGoToDefinitionService.cs

```diff
@@ -111,38 +119,34 @@ public bool TryGoToDefinition(Document document, int position, CancellationToken
             // In the future we can expand this with other mappings if appropriate.
             var interfaceImpls = symbol.ExplicitOrImplicitInterfaceImplementations();
             if (interfaceImpls.Length == 0)
-                return false;
+                return null;
 
             var title = string.Format(EditorFeaturesResources._0_implemented_members,
                 FindUsagesHelpers.GetDisplayName(symbol));
 
-            return _threadingContext.JoinableTaskFactory.Run(async () =>
+            using var _ = ArrayBuilder<DefinitionItem>.GetInstance(out var builder);
+            foreach (var impl in interfaceImpls)
             {
-                using var _ = ArrayBuilder<DefinitionItem>.GetInstance(out var definitions);
-                foreach (var impl in interfaceImpls)
-                {
-                    // Use ConfigureAwait(true) here.  Not for a correctness requirements, but because we're
-                    // already blocking the UI thread by being in a JTF.Run call.  So we might as well try to
-                    // continue to use the blocking UI thread to do as much work as possible instead of making
-                    // it wait for threadpool threads to be available to process the work.
-                    definitions.AddRange(await GoToDefinitionHelpers.GetDefinitionsAsync(
-                        impl, solution, thirdPartyNavigationAllowed: false, cancellationToken).ConfigureAwait(true));
-                }
+                builder.AddRange(await GoToDefinitionHelpers.GetDefinitionsAsync(
+                    impl, solution, thirdPartyNavigationAllowed: false, cancellationToken).ConfigureAwait(false));
+            }
+
+            var definitions = builder.ToImmutable();
 
-                return await _streamingPresenter.TryNavigateToOrPresentItemsAsync(
-                    _threadingContext, solution.Workspace, title, definitions.ToImmutable(), cancellationToken).ConfigureAwait(true);
-            });
+            return await _streamingPresenter.GetNavigableLocationAsync(
+                _threadingContext, solution.Workspace, title, definitions, cancellationToken).ConfigureAwait(false);
```

Comment:
Seems a bit odd this is a method on the streaming presenter here?

Reply:
Ah: I see it's also displaying the items too; should this still be called PresentAndGetItems? Or what happens if we're presenting and then nothing further is shown?

Reply:
yeah... this guy is funky in that it gets back an object that can optionally navigate you to a since location (if there is one) or which actually displayes the presenter if there are many.  Importantly though, that presentation/navigation only happens on the stage-2 of the navigation.  will think on a better name here.